pkg/progress/progress.go

```diff
@@ -83,18 +82,28 @@ func (m *Manager) UpdateProgressRemaining(progress string, remaining float64) {
 		if p.total < remaining {
 			p.total = remaining
 		}
-		if p.lastTimeRemaining < remaining {
-			p.lastTimeRemaining = remaining
+
+		// We use window size / update interval to get the history length.
+		historyLen := int(speedStatisticalWindow / updateInterval)
```

Comment:
Could we set `historyLen` as a constant(an attribute for `Manager`) once `Manager` created?